### ISSUE TITLE
Duplicate warnings for match_bool / single_match

### ISSUE THREAD

**Issue by llogiq** [Manishearth/rust-clippy#403]
The following snippet generates two lint warnings:

```
match test { //~ ERROR you seem to be trying to match on a boolean expression
    //~^ERROR you seem to be trying to use match
    false => { println!("Noooo!"); },
    _ => (),
};
```

FWIW, I think the right option is to skip the `single_match` lint if the `match_bool` lint already warned.

**Comment by wartman4404** [comment]
I'd like to take a shot at this.
I agree, `match_bool` seems like it's more specific, and should take precedence.

**Comment by wartman4404** [comment]
How should this interact with lint levels?  If `match_bool` is set to Allow, `single_match` should still display a warning, but what if `single_match` is Deny and `match_bool` is Warn?

**Comment by Manishearth** [comment]
I'm fine with unconditionally not running the let lint when the match is a bool match.

Adding special handling for when the bool match is Allow works too. I don't think we need to handle all mixes of the lints, if both are visible then showing at least one is good enough.

**Comment by wartman4404** [comment]
Sounds good.  I guess I was overthinking it.

**Comment by Manishearth** [comment]
#403
